src/rdkafka_cgrp.c

```diff
@@ -2925,13 +2925,25 @@ rd_kafka_cgrp_consumer_assignment_with_metadata(
                 rd_kafka_Uuid_t request_topic_id =
                     rd_kafka_topic_partition_get_topic_id(
                         &assignment->elems[i]);
+                int partition             = assignment->elems[i].partition;
+                rd_bool_t partition_found = rd_false;
 
                 rd_kafka_rdlock(rk);
                 rkmce =
                     rd_kafka_metadata_cache_find_by_id(rk, request_topic_id, 1);
 
-                if (rkmce)
+                if (rkmce) {
                         topic_name = rd_strdup(rkmce->rkmce_mtopic.topic);
+                        // Check if partition exists in metadata
+                        int pcnt = rkmce->rkmce_mtopic.partition_cnt;
+                        for (j = 0; j < pcnt; j++) {
+                                if (rkmce->rkmce_mtopic.partitions[j].id ==
+                                    partition) {
+                                        partition_found = rd_true;
+                                        break;
+                                }
+                        }
```

Comment:
Since the partitions are monotonically increasing and cannot be deleted in Kafka, we can just use `cnt` to know if the partition is present in the metadata or not.


```suggestion
                        partition_found = (partition < pcnt);
```